lib/taskana-core/src/test/java/acceptance/config/TaskanaSecurityConfigAccTest.java

```diff
@@ -0,0 +1,134 @@
+
+import pro.taskana.TaskanaEngineConfiguration;
+import pro.taskana.common.api.exceptions.SystemException;
+import pro.taskana.common.internal.TaskanaEngineTestConfiguration;
+import pro.taskana.common.internal.configuration.DbSchemaCreator;
+import pro.taskana.sampledata.SampleDataGenerator;
+
+public class TaskanaSecurityConfigAccTest {
+
+  @BeforeAll
+  public static void setupTests() throws SQLException {
+    DataSource dataSource = TaskanaEngineTestConfiguration.getDataSource();
+    String schemaName = TaskanaEngineTestConfiguration.getSchemaName();
+    DbSchemaCreator dbSchemaCreator = new DbSchemaCreator(dataSource, schemaName);
+    dbSchemaCreator.run();
+  }
+
+  @AfterEach
+  public void cleanDb() {
+    DataSource dataSource = TaskanaEngineTestConfiguration.getDataSource();
+    String schemaName = TaskanaEngineTestConfiguration.getSchemaName();
+
+    SampleDataGenerator sampleDataGenerator = new SampleDataGenerator(dataSource, schemaName);
+    sampleDataGenerator.clearDb();
+  }
+
+  @Test
+  public void should_ThrowException_When_CreatingUnsecuredEngineConfigWhileSecurityIsEnforced()
+      throws SQLException {
+
+    setSecurityFlag(true);
+
+    ThrowingCallable createUnsecuredTaskanaEngineConfiguration = () -> {
+
+      TaskanaEngineConfiguration taskanaEngineConfiguration = new TaskanaEngineConfiguration(
+          TaskanaEngineTestConfiguration.getDataSource(), false, false,
+          TaskanaEngineTestConfiguration.getSchemaName());
+    };
+
+    assertThatThrownBy(createUnsecuredTaskanaEngineConfiguration)
+        .isInstanceOf(SystemException.class)
+        .hasMessageContaining("Secured TASKANA mode is enforced, can't start in unsecured mode");
+
+  }
+
+  @Test
+  public void should_StartUpNormally_When_CreatingUnsecuredEngineConfigWhileSecurityIsNotEnforced()
+      throws SQLException {
+
+    setSecurityFlag(false);
+
+    ThrowingCallable createUnsecuredTaskanaEngineConfiguration = () -> {
+
+      TaskanaEngineConfiguration taskanaEngineConfiguration = new TaskanaEngineConfiguration(
+          TaskanaEngineTestConfiguration.getDataSource(), false, false,
+          TaskanaEngineTestConfiguration.getSchemaName());
+    };
+
+    assertThatCode(createUnsecuredTaskanaEngineConfiguration).doesNotThrowAnyException();
+
+  }
+
+  @Test
+  public void should_SetSecurityFlag_When_SecurityFlagIsNotSet() throws SQLException {
+
+    assertThat(retrieveSecurityFlag()).isNull();
+
+    ThrowingCallable createUnsecuredTaskanaEngineConfiguration = () -> {
+
+      TaskanaEngineConfiguration taskanaEngineConfiguration = new TaskanaEngineConfiguration(
+          TaskanaEngineTestConfiguration.getDataSource(), false, false,
+          TaskanaEngineTestConfiguration.getSchemaName());
+    };
+
+    assertThatCode(createUnsecuredTaskanaEngineConfiguration).doesNotThrowAnyException();
+
+    assertThat(retrieveSecurityFlag()).isFalse();
+
+  }
+
+  private Boolean retrieveSecurityFlag() throws SQLException {
```

Comment:
According to my comment above: do we miss the test for starting secure the first time and initialize the security flag?